.github/workflows/master-pr-build.yaml

```diff
@@ -0,0 +1,68 @@
+## ---------------------------------------------------------------------------
+## Licensed to the Apache Software Foundation (ASF) under one or more
+## contributor license agreements.  See the NOTICE file distributed with
+## this work for additional information regarding copyright ownership.
+## The ASF licenses this file to You under the Apache License, Version 2.0
+## (the "License"); you may not use this file except in compliance with
+## the License.  You may obtain a copy of the License at
+##
+##      http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+## ---------------------------------------------------------------------------
+
+name: master pr build
+
+on:
+  pull_request:
+    branches:
+      - master
+
+env:
+  MAVEN_OPTS: -Xmx4g -Xms1g -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.http.retryHandler.count=3
+
+jobs:
+  mvn:
+    timeout-minutes: 180
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        java: [ '1.8' ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache Maven Repository
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository/
+            !~/.m2/repository/org/apache/servicecomb
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Show the first log message
+        run: git log -n1
+      - name: Build and test
+        if: ${{ success() }}
+        run: ./mvnw clean install -B -Pjacoco -Pdocker coveralls:report -DrepoToken=ftogh57jTQ0GRa4AaOgFfltiUrwnO8tw9
```

Comment:
The token string should be used in environment variables ?

Reply:
> The token string should be used in environment variables ?

secrets are not passed to the runner when a workflow is triggered from a forked repository.

As GitHub stuff answered [here](https://github.community/t/make-secrets-available-to-builds-of-forks/16166/14)

Reply:
OK, I see. so is there any other problem we provide the token in plain  text here ? this token could be changed in the future ?

Reply:
The token has been set to GitHub secrets with https://issues.apache.org/jira/browse/INFRA-21569 and uses secrets.COVERALLS_TOKEN in push workflow. 

If we do not analyze the coverage in the pull request, we can remove the plaintext token

Reply:
hmm, I prefer to disable the coverage in PR now and see if we can have a better solution in the future. How do you think @WillemJiang

Reply:
> hmm, I prefer to disable the coverage in PR now and see if we can have a better solution in the future. How do you think @WillemJiang

I removed the Coveralls report token from the pull request script